### DISCUSSION TITLE
unsupported arch armv7l on raspberry pi 4 when attempting to install LDC

### DISCUSSION THREAD

**Discussion by prizzledev** [abraunegg/onedrive#1423] [Q&A] [answered]
If I run `curl -fss https://dlang.org/install.sh | bash -s ldc `on my raspberry pi 4 in terminal, I get
`unsupported arch armv7l` as respond. I'm very new at Linux and Raspberry, so please describe exactly what I have to do.
(Maybe I don't even need it, because it is not in the INSTALL.md for my system. I thought I need it, because I just followed this tutorial[here](https://youtu.be/abwW3En7qOQ))

Thank you <3

**Comment by abraunegg** [comment, marked as answer]
@Prizzledizle 
> If I run `curl -fss https://dlang.org/install.sh | bash -s ldc` on my raspberry pi 4 in terminal, I get
`unsupported arch armv7l` as respond.

This is because LDC cannot be installed that way on a Raspberry Pi.

For a Raspberry PI you *must* follow these instructions:
* https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#dependencies-raspbian-armhf
* https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#dependencies-ubuntu-20x--debian-10-arm64

Given you are running `armv7l` you need to follow the ARMHF instructions.
